packages/util/test/pipe.spec.ts

```diff
@@ -0,0 +1,20 @@
+import { identity, pipe } from '../src';
+
+describe('pipe', () => {
+  const addAlice = (str: string): string => str + 'Alice';
+  const addBob = (str: string): string => str + 'Bob';
+  const addCharlie = (str?: string): string =>
+    str ? str + 'Charlie' : 'noCharlie';
+
+  it('should execute sync pipe with argument', () => {
```

Comment:
I think the original `pipe` supports the type check of the parameters.
Could you please add a test for example:
```
pipe(
        () => {return 1},
        (s: string)=> {return ''}
      )();
```

Reply:
I was breaking my head how to do that properly and checked finally in the `index.d.ts` of the rambda and found explicit typing like: 
```
export function pipe<V0, T1, T2, T3, T4, T5, T6, T7, T8>(
  fn0: (x: V0) => T1,
  fn1: (x: T1) => T2,
  fn2: (x: T2) => T3,
  fn3: (x: T3) => T4,
  fn4: (x: T4) => T5,
  fn5: (x: T5) => T6,
  fn6: (x: T6) => T7,
  fn: (x: T7) => T8): (x: V0) => T8;
```

and this for all funcitons from one to 10 arguments. So this is how they achieve type safety (up to ten arguments). I can put this but then we have multiple overloaded implementations.

Reply:
ok, you can also argue, the home made pipe function doesn't support the type safe in a chain.
In this way, maybe git rid of pipe? because the vanilla code is safer?
Or we check all the usage in the sdk, if they only handle functions like `(string) => string`, we can also limit the signature of the home made pipe.

Reply:
I removed the pipe.